Allow override onProgress on Upload

minwe opened the pull request:
allow user to disable upload progress listener

@see https://github.com/react-component/upload/issues/33#issuecomment-284644039

and https://github.com/react-component/upload/pull/65

afc163 commented:
```diff
- assign({}, this.props, {
-   onStart: this.onStart,
-   onError: this.onError,
-   onProgress: this.onProgress
-   onSuccess: this.onSuccess,
- });
+ assign({}, {
+  onStart: this.onStart,
+  onError: this.onError,
+  onProgress: this.onProgress
+  onSuccess: this.onSuccess,
+ }, this.props);
```

Just switch props order so you can set `<Upload onProgress={null} />` for your very special situation. `listenUploadProgress` is a prop hard to understand.

afc163 commented:
<img width="319" alt="image" src="https://cloud.githubusercontent.com/assets/507615/23789224/dd1c2d4a-05b4-11e7-892d-d9e00e235ee2.png">

minwe committed:
feat: allow override onProgress on Upload

@see https://github.com/react-component/upload/issues/33#issuecomment-284644039

minwe commented:
@afc163 changed and rebased

afc163 commented:
PR to master

afc163 commented:
rebase master again, there is still a extra commit.

minwe commented:
Rebase done.
